### ISSUE TITLE
Performance

### ISSUE THREAD

**Issue by asthomas** [bridgedotnet/Granular#31]
I tried playing with granularpad in both Firefox and Chrome.  I notice that it takes about 3 seconds to load some of the examples, like the Transforms and the KeyboardNavigation examples.  The KeyboardNavigation example has 58 controls on it.  I am looking at creating an app that has about 10000 controls on the screen.  Does performance scale linearly in Granular?  Should I expect the load time on the page to be about 8 minutes for my application?

**Comment by yuvaltz** [comment]
Hi,

The next version (0.2.1) which I hope to release soon, will have some major performance improvements.
In the current version (0.2.0) some of the performance issues are caused by inefficient processes, such as eager resources loading or redundant layout passes, which are some of the things I've already solved.

I'm not sure about scaling to thousands of controls (definitely with the default templates which are very wasteful), I'll try to investigate it and see if there is something I could do for this release.
I don't think the browser (or anyone else) will wait 8 minutes for an application to load, so unfortunately it's impossible to create such large applications as it is right now.

I don't see however any reason why a great performance couldn't be achieved, even for large applications, I think that the only thing stopping Granular from getting there is [too few eyeballs on it](https://en.wikipedia.org/wiki/Linus's_Law) (mines only, currently).

In the current version (0.2.0) I've migrated Granular to Bridge, and improved the compatibility with WPF (by ProjectLink build task, and other features), now I'm trying to achieve a somewhat more reasonable performance, hoping to attract new developers that may help to find more ways to improve it even further.
